src/Test/L0/Worker/VariablesL0.cs

```diff
@@ -755,6 +755,51 @@ public void Set_StoresValue()
             }
         }
 
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void IsReadOnly_RespectsSystemVariables()
+        {
+            using (TestHostContext hc = new TestHostContext(this))
+            {
+                // Arrange.
+                List<string> warnings;
+                var variables = new Variables(hc, new Dictionary<string, VariableValue>(), out warnings);
+                variables.Set(Constants.Variables.Agent.ReadOnlyVariables, "true");
```

Comment:
should we have some tests for when this is set to "false"?

Reply:
Yep, added